posts/python-in-production-part-5-of-5.md

```diff
@@ -0,0 +1,250 @@
+    become: true
+    apt:
+      update_cache: yes
+      state: latest
+      name:
+      - apt-transport-https
+      - ca-certificates
+      - curl
+      - gpg-agent
+      - software-properties-common
+
+  - name: add docker GPG key
+    become: true
+    apt_key:
+      url: https://download.docker.com/linux/debian/gpg
+      state: present
+
+  - name: add docker apt repo
+    become: true
+    apt_repository:
+      repo: deb https://download.docker.com/linux/debian bullseye stable
+      state: present
+
+  - name: install docker
+    become: true
+    apt:
+      update_cache: yes
+      state: latest
+      name:
+      - docker-ce
+      - docker-ce-cli
+      - containerd.io
+
+  ## Setup Ansible to Control Docker
+
+  - name: install python dependencies
+    become: true
+    apt:
+      update_cache: yes
+      state: latest
+      name: python3-pip
+
+  - name: install 'Docker SDK for Python'
+    pip:
+      name: docker
+
+  ## Build Docker Image
+  - name: create build directory
+    file:
+      path: /root/jenkins-dockerfile
+      state: directory
+      owner: root
+      group: root
+      mode: '0755'
+  - name: copy Dockerfile
+    copy:
+      src: ./Dockerfile
+      dest: /root/jenkins-dockerfile/Dockerfile
+      owner: root
+      group: root
+      mode: '0644'
+  - name: build container image
+    docker_image:
+      name: jenkins_with_python
+      source: build
+      build:
+        path: /root/jenkins-dockerfile
+      state: present
+
+  ## Start Jenkins
+  - name: Start Jenkins
+    docker_container:
+      name: jenkins
+      image: jenkins_with_python
+      state: started
+      ports:
+       - "8080:8080"
+       - "50000:50000"
+      volumes:
+       - jenkins_home:/var/jenkins_home
+```
+
+Run this playbook by typing: `ansible-playbook -i hosts jenkins.yml --extra-vars "ansible_user={{your_vm_user}}"`.  Be sure to put your username that you use to SSH with into the `your_vm_user` placeholder spot (without the doulbe curly braces).  For more information on ansible, visit there awsome [documentation](https://docs.ansible.com/ansible/latest/getting_started/index.html).
+
+# The End
+
+If you made it all the way to the end of this series, from the beginning, then you have just finished a massive journey through a ton of awesome tech, and tools.  In this series, we covered the how's and why's of:
+* Creating a Python virtual environment in [part 1](https://templinkplaceholder).
+* Using Python Modules and Packages in [part 2](https://templinkplaceholder).
+* Building a Single File Binary in [part 3](https://templinkplaceholder).
+* Running your Program as a Linux Service using Systemd in [part 4](https://templinkplaceholder).
+* Automating your Build Process with Jenkins in [part 5](https://templinkplaceholder).
+
+If you are a Software Engineer or a DevOps Engineer and want to implement some of these practices in your organization but need a little help, feel free to reach out to us!
```

Comment:
```suggestion
If you are a Software Engineer or a DevOps Engineer and want to implement some of these practices in your organization, Ippon Technologies can help! Contact [sales@ipponusa.com](mailto:sales@ipponusa.com) with any questions.
```

Reply:
Just gave some additional color to the call to action and included the sales email